src/psij/job_state.py

```diff
@@ -98,3 +98,33 @@ def __eq__(self, other: object) -> bool:
             return bool(self._value_ == other._value_)
         else:
             return False
+
+    def __ne__(self, other: object) -> bool:
+        """Returns false only if `other` represents the same state as this state."""
+        if isinstance(other, JobState):
+            return bool(self._value_ != other._value_)
+        else:
+            return True
+
+    def __bool__(self) -> bool:
+        """All states are consider true-ish."""
+        return True
+
+
+class JobStateOrder:
+    """A class that can be used to reconstruct missing states."""
+
+    @staticmethod
+    def prev(state: JobState) -> Optional[JobState]:
+        """Returns the state previous to the given state.
+
+        The "previous" state is a state that must have occurred immediately prior to this state
+        given the state transition diagram if such a state is unique. Not all states have a
+        previous state. For example, the FAILED state does not have a previous state, since it can
+        be reached from multiple states.
+        """
+        if state == JobState.COMPLETED:
+            return JobState.ACTIVE
+        if state == JobState.ACTIVE:
+            return JobState.QUEUED
+        return None
```

Comment:
maybe add 
```py
        if state == JobState.QUEUED:
            return JobState.NEW
```
?

Reply:
True but unnecessary since the job starts in the NEW state.

Reply:
I am not sure I understand the comment.  Yes, the job starts in `NEW`, but (looking at the doc string) the method is supposed to give the previous state to any state which has a previous state, and `QUEUED` has the previous state `NEW`.  So why is that unnecessary?  Are there assumptions on where/when that method can be used?

Reply:
It's only used to fill in missing states in _set_status, and NEW is never missing.

Reply:
I would like to see it added anyway: otherwise it would need a docstring change to document that it cannot be called on `QUEUED` which seems more effort than adding it.  It's a public method after all.